.github/CONTRIBUTING.md

```diff
@@ -67,6 +67,99 @@ See vignette [function naming logic](https://cynkra.github.io/dm/articles/tech-d
 
+## Test databases
+
+This repository comes with a `docker-compose.yml` file that sets up the databases required for testing.
+All shell commands are expected to be run from the top-level directory of a clone of this repository.
+
+### macOS only: Install colima to run Docker containers
+
+```sh
+brew install colima docker-compose
+colima start -c 4 -m 4 --vm-type vz --vz-rosetta --network-address
+colima status
+#  INFO[0000] colima is running using macOS Virtualization.Framework
+#  INFO[0000] arch: aarch64
+#  INFO[0000] runtime: docker
+#  INFO[0000] mountType: virtiofs
+#  INFO[0000] address: 192.168.64.2
+#  INFO[0000] socket: unix:///Users/kirill/.colima/default/docker.sock
+```
+
+Take note of the `address` in the output, it will be used later.
+
+See also <https://docs.google.com/document/d/1axInaYK6oK6riRio72uTAeQazuork1X0clY9UL9gYoE/edit?usp=sharing> for more details on colima.
+
+### mssql: ODBC drivers
+
+- Linux: <https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver16&tabs=ubuntu18-install%2Calpine17-install%2Cdebian8-install%2Credhat7-13-install%2Crhel7-offline>
+
+- macOS: <https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/install-microsoft-odbc-driver-sql-server-macos?view=sql-server-ver16>
+
+### Start the containers
+
+```sh
+docker-compose up
+# May take several minutes to pull the images
+```
+
+Switches:
+
+- `--force-recreate`: Recreate all databases from scratch
+- `-d`: Daemon mode, run in the background, use `docker-compose stop` to stop
+- `maria`, `mssql`, `postgres`: Run only a particular service
+
+### Test against a specific database backend
+
+macOS:
+
+```sh
+DM_TEST_SRC=mssql DM_TEST_DOCKER_HOST=192.168.64.2 R -q -e 'testthat::test_local()'
+```
+
+Linux:
+
+```sh
+DM_TEST_SRC=mssql DM_TEST_DOCKER_HOST=localhost DM_TEST_MSSQL_ODBC_LIB=/opt/microsoft/msodbcsql18/lib64/libmsodbcsql-18.2.so.1.1 R -q -e 'testthat::test_local()'
+```
+
+Controlled with environment variables:
+
+- `DM_TEST_SRC`: backend, one of:
+
+    - `df` (default)
+    
+    - `sqlite`
+    
+    - `duckdb`
+    
+    - `postgres`
+    
+    - `maria`
+    
+    - `mssql` (needs to create a database, see below)
+
+- `DM_TEST_DOCKER_HOST`: `localhost` on Linux, see output of `colima status` on macOS
+
+- `DM_TEST_MSSQL_ODBC_LIB`: mssql, path to ODBC library (`libmsodbcsql-*.so` or similar), the default should work for Homebrew-installed drivers on macOS arm64
+
+### mssql: For a new container, create the database
+
+FIXME: Automate this.
+
+macOS:
+
+```sh
+DM_TEST_DOCKER_HOST=192.168.64.2 R -q -e 'pkgload::load_all(); DBI::dbExecute(test_src_mssql(FALSE)$con, "CREATE DATABASE test")'
+```
```

Comment:
This did not work in the terminal for me:
```
tobias@Tobiass-MacBook-Pro dm % DM_TEST_DOCKER_HOST=192.168.64.2 R -q -e 'pkgload::load_all(); DBI::dbExecute(test_src_mssql(FALSE)$con, "CREATE DATABASE test")'
WARNING: ignoring environment value of R_HOME
> pkgload::load_all(); DBI::dbExecute(test_src_mssql(FALSE)$con, "CREATE DATABASE test")

 *** caught segfault ***
address 0x18, cause 'invalid permissions'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 4: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 5: loadNamespace(x)
An irrecoverable exception occurred. R is aborting now ...
zsh: segmentation fault  DM_TEST_DOCKER_HOST=192.168.64.2 R -q -e 
```

doing the same in R worked though (I set the necessary environment variables beforehand):
```
> src_mssql <- test_src_mssql(FALSE)
> DBI::dbExecute(src_mssql$con, "CREATE DATABASE test") 
[1] 0
```

Reply:
Could this be unrelated trouble with the R installation?